### PR TITLE
HPCC-29788 Make multiJobLinger=true the default.

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - I'm targetting 9.0 here.. but perhaps this change should go into 9.2 ?

File: thorlcr/master/thgraphmanager.cpp [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
             else
             {
                 unsigned lingerPeriod = globals->getPropInt("@lingerPeriod", DEFAULT_LINGER_SECS)*1000;
```
Comment (ghalliday) [inline thorlcr/master/thgraphmanager.cpp:1330]: Is the default linger time suitable?

File: thorlcr/master/thgraphmanager.cpp [hunk 2]
```diff
@@ -1328,7 +1328,7 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
...
-                bool multiJobLinger = globals->getPropBool("@multiJobLinger");
+                bool multiJobLinger = globals->getPropBool("@multiJobLinger", true);
```
Comment (ghalliday) [inline thorlcr/master/thgraphmanager.cpp:1331]: What will be the possible effect on mixed version environments?  will it fall-back gracefully?

**Review by ghalliday — APPROVED** [review]
Looks good.  A couple of questions.

File: thorlcr/master/thgraphmanager.cpp (hunk 1 above)
Comment (jakesmith) [reply thorlcr/master/thgraphmanager.cpp:1330]: for a contested cluster it is.. e.g. inter-leaved graphs from queues jobs
would be enough e.g. for regression suite.

I've upped default to 1 min

File: thorlcr/master/thgraphmanager.cpp (hunk 2 above)
Comment (jakesmith) [reply thorlcr/master/thgraphmanager.cpp:1331]: not quite sure what you mean - do you mean if agent was different to thor?

The setting is generated in to the agent settings from the thor: settings.

**Comment by jakesmith** [comment]
@ghalliday - please see replies.

**Review by ghalliday — APPROVED** [review]

**Comment by ghalliday** [comment]
@jakesmith @JamesDeFabia where does this change in behaviour need documenting?

**Comment by jakesmith** [comment]
I'll defer to Jim. @JamesDeFabia  Is there a general chapter/paragraph regarding k8s on-demand components?

**Pull request merged**